### PR TITLE
sample UI test case for explicit login

### PR DESCRIPTION
**Pull request by pratap511**
- sample UI test case for explicit login

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 source 'https://github.com/CocoaPods/Specs.git'
 
-platform :ios, '9.0'
+platform :ios, '13.0'
```
Comment (zakirjt) [inline Podfile]: why is it iOS 13 ? We are supporting only 14 and 15.

File: Podfile.lock [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+  - MASProximity (2.1.00):
+    - MASFoundation (= 2.1.00)
+  - MASUI (2.1.00):
+    - MASFoundation (= 2.1.00)
+    - MASProximity (= 2.1.00)
 
 DEPENDENCIES:
-  - MASFoundation (~> 1.9.10)
-  - MASUI (~> 1.9.10)
+  - MASFoundation (~> 2.1.00)
```
Comment (zakirjt) [inline Podfile.lock]: Versions at all places now should be changed to 2.2

File: msso_config.json [hunk 3]
```diff
@@ -1 +1,136 @@
-Add msso_config from your server here
```
Comment (zakirjt) [inline msso_config.json:1]: Should we add the msso config server details here ?
@ysd24 , please guide.

**Review by as666499 — DISMISSED** [review]

**Comment by ysd24** [comment]
@pratap511  I see one of the test 'test_InvokeAPILogin' is failing. Could you reverify if the 'okButtonLoginSuccess' is being evaluated correctly. Looks like its trying to find the username/password textfield. Please test thoroughly.

**Comment by pratap511** [comment]
> @pratap511 I see one of the test 'test_InvokeAPILogin' is failing. Could you reverify if the 'okButtonLoginSuccess' is being evaluated correctly. Looks like its trying to find the username/password textfield. Please test thoroughly.

Working on it

File: Podfile (hunk 1 above)
Comment (pratap511) [reply Podfile]: Will change version accordingly

File: Podfile.lock (hunk 2 above)
Comment (pratap511) [reply Podfile.lock]: Will changed as per latest update

File: msso_config.json (hunk 3 above)
Comment (pratap511) [reply msso_config.json:1]: if it's not require will clean this file

Comment (ysd24) [reply msso_config.json:1]: @pratap511 We do not checkin the msso_config files. Please leave them blank.

Comment (pratap511) [reply msso_config.json:1]: Sure @ysd24

File: MASAuthenticationUITests/MASAuthenticationUITests.m [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+#define INVOKING_APISUCCESS_MESSAGE       @"Invoking API Successful"
+#define Authentication_SUCCESS            @"Authentication status: authenticated as"
+#define DE_REGISTER_DEVICE_MESSAGE        @"Authentication status: not authenticated"
+#define SYSTEM_ALERT_HANDLER              @"SystemAlertHandler"
+#define ALLOW_ONCE                        @"Allow Once"
+#define TIME_INTERVAL                     10
+#define MAS_UI_CANCEL                     @"masui-cancelBtn"
+#define MAS_UI_LogIn                      @"Login"
+#define MAS_UI_USER_TEXTFIELD             @"masui-usernameField"
+#define MAS_UI_PASSWORD_FIELD             @"masui-passwordField"
```
Comment (ysd24) [inline MASAuthenticationUITests/MASAuthenticationUITests.m:29]: Please maintain a separate json file and take input from there. Also if possible don't expose any sensitive data like username/passwords as these sample apps will get shared with customers too.

File: MASAuthentication.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -236,18 +301,20 @@
 			files = (
 			);
 			inputPaths = (
-				"${SRCROOT}/Pods/Target Support Files/Pods-MASAuthentication/Pods-MASAuthentication-frameworks.sh",
+				"${PODS_ROOT}/Target Support Files/Pods-MASAuthentication/Pods-MASAuthentication-frameworks.sh",
```
Comment (ysd24) [inline MASAuthentication.xcodeproj/project.pbxproj:308]: Do not checkin the xcworkspace changes created by cocoapods. Please revert back to old state. Add your UITests files and checkin only the relevant xcodeproj/pbxproj changes.

File: MASAuthentication.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ysd24) [inline MASAuthentication.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Do not checkin this file.

File: MASAuthenticationUITests/MASAuthenticationUITests.m [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        [passwordElement typeText:PASSWORD];
+    }
+}
+-(NSString *)succeessUser {
+    return [NSString stringWithFormat:@"%@ %@",Authentication_SUCCESS,USER_NAME];
+}
+
+- (void)test_ExplicitLogIn {
+    
+    [_app launch];
```
Comment (ysd24) [inline MASAuthenticationUITests/MASAuthenticationUITests.m:110]: Please add proper comments - There are not comments at all. We can not make out what each step is trying to do.
Also please follow our commenting style for example : 
//
// Invoke protected endpoint
//

File: MASAuthenticationUITests/MASAuthenticationUITests.m [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+    [_app.staticTexts[DE_REGISTER_DEVICE] tap];
+    
+    XCUIElement *deRegisterDevice = _app.staticTexts[DE_REGISTER_DEVICE_MESSAGE];
+    
+    NSPredicate *exists = [NSPredicate predicateWithFormat:@"exists == 1"];
+    [self expectationForPredicate:exists evaluatedWithObject:deRegisterDevice handler:nil];
+    [self waitForExpectationsWithTimeout:TIME_INTERVAL handler:nil];
+    
+    XCTAssert([deRegisterDevice exists]);
+}
```
Comment (ysd24) [inline MASAuthenticationUITests/MASAuthenticationUITests.m:323]: Please add negative test scenarios. I don't see a invalid login credentials scenario is any of the above test cases.

**Comment by pratap511** [comment]
> @pratap511 I see one of the test 'test_InvokeAPILogin' is failing. Could you reverify if the 'okButtonLoginSuccess' is being evaluated correctly. Looks like its trying to find the username/password textfield. Please test thoroughly.

modified test case accordingly

**Review by ysd24 — APPROVED** [review]
Looks good.

**Pull request merged**